torch/_dynamo/variables/builder.py

```diff
@@ -1565,7 +1565,29 @@ def __eq__(self, other: object) -> bool:
 
 # Performs automatic dynamic dim determination.
 # Returns a CreateSymbolicPolicy
-def _automatic_dynamic(e, tx, name, static_shapes) -> CreateSymbolicPolicy:
+def _automatic_dynamic(
+    e, tx, name, static_shapes, recurse=True
+) -> CreateSymbolicPolicy:
+    is_subclass = is_traceable_wrapper_subclass(e)
+    if recurse and is_subclass:
+        outer_policy = _automatic_dynamic(e, tx, name, static_shapes, recurse=False)
```

Comment:
I don't understand why its 1 layer? It should always recurse, until we hit a proper exit condition, right?

Reply:
For a given subclass input `e`, we want to run automatic dynamic on both the outer shape and inner tensor shapes. The `recurse=False` trick is for running the algo on the outer shape only.

I believe it will recurse as needed if we have inner tensors that are subclasses.